sdk/azidentity/azure_cli_credential.go

```diff
@@ -0,0 +1,177 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azidentity
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"regexp"
+	"runtime"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+)
+
+type AzureCLITokenProvider interface {
+	GetCLIToken(ctx context.Context, resource string) ([]byte, error)
+}
+
+type azureCLITokenProvider struct{}
+
+// AzureCLICredentialOptions contains options used to configure the AzureCLICredential
+type AzureCLICredentialOptions struct {
+	TokenProvider AzureCLITokenProvider
+}
+
+// AzureCLICredential enables authentication to Azure Active Directory using the Azure CLI command "az account get-access-token".
+type AzureCLICredential struct {
+	tokenProvider AzureCLITokenProvider
+}
+
+// TODO: do we want an options bag for this credential that includes a developer specified Azure CLI path or only specify with env var?
+// NewAzureCLICredential constructs a new AzureCLICredential with the details needed to authenticate against Azure Active Directory
+// options: configure the management of the requests sent to Azure Active Directory.
+func NewAzureCLICredential(options *AzureCLICredentialOptions) (*AzureCLICredential, error) {
+	if options != nil {
+		return &AzureCLICredential{
+			tokenProvider: options.TokenProvider,
```

Comment:
maybe this would be better with a setTokenProvider() to check that the token provider is not empty?